File: sample-videochat-java/app/src/main/java/com/quickblox/sample/videochat/java/utils/UsersUtils.java
package com.quickblox.sample.videochat.java.utils;

import com.quickblox.sample.videochat.java.db.UsersDbManager;
import com.quickblox.users.model.QBUser;

import java.util.ArrayList;
import java.util.List;

public class UsersUtils {
    private static SharedPrefsHelper sharedPrefsHelper;
    private static UsersDbManager dbManager;

    public static ArrayList<QBUser> getUsersFromIds(ArrayList<QBUser> existedUsers, List<Integer> allIds) {
        ArrayList<QBUser> users = new ArrayList<>();
        for (Integer userId : allIds) {
            QBUser stubUser = createStubUserById(userId);
            if (!existedUsers.contains(stubUser)) {
                users.add(stubUser);
            }
        }
        users.addAll(existedUsers);

        return users;
    }

    private static QBUser createStubUserById(Integer userId) {
        QBUser stubUser = new QBUser(userId);
        stubUser.setFullName(String.valueOf(userId));
        return stubUser;
    }

    public static ArrayList<Integer> getIdsNotLoadedUsers(ArrayList<QBUser> existedUsers, List<Integer> allIds) {
        ArrayList<Integer> idsNotLoadedUsers = new ArrayList<>();
        for (Integer userId : allIds) {
            QBUser stubUser = createStubUserById(userId);
            if (!existedUsers.contains(stubUser)) {
                idsNotLoadedUsers.add(userId);
            }
        }

        return idsNotLoadedUsers;
    }

    public static void removeUserData() {
        if (sharedPrefsHelper == null) {
            sharedPrefsHelper = SharedPrefsHelper.getInstance();
        }
        sharedPrefsHelper.clearAllData();
        if (dbManager == null) {
            dbManager = UsersDbManager.getInstance();
        }
        dbManager.clearDB();
    }
}

File: sample-videochat-kotlin/src/main/java/com/quickblox/sample/videochatkotlin/utils/UsersUtils.kt
package com.quickblox.sample.videochatkotlin.utils

import com.quickblox.users.model.QBUser
import java.util.ArrayList

fun getListAllUsersFromIds(opponentsIds: List<Int>): ArrayList<QBUser> {
    val qbUsers = ArrayList<QBUser>()

    for (userId in opponentsIds) {
        val stubUser = createStubUserById(userId)
        qbUsers.add(stubUser)
    }

    return qbUsers
}

private fun createStubUserById(userId: Int?): QBUser {
    val stubUser = QBUser(userId)
    stubUser.fullName = userId.toString()
    return stubUser
}